### PR TITLE
endpoint to log latefee

### PR DESCRIPTION
**Pull request by darmhoo**
#### What does this PR do?
- (replace me) Explain WHY this PR is required. E.g.: What feature it adds, what problem it solves, …

#### Description of Task proposed in this pull request?
- (replace me) A description of WHAT changed in the codebase. Additionally add the reasoning if it's complex or abstract. 

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [Story ID:  replace text with story id](paste Pivotal Tracker link here)

#### Any background context you want to add (Operations Impact)?
- (replace me) List the steps that must be taken for this PR to work.
E.g.: `npm install`, `Add "github_key" to environment variables, ...`

Be sure to include all systems that needs to be changed or which system is
affected by the change (Ex: Requires Styled Components to be installed and
configured in `babel.rc`, file to be updated).

#### What I have learned working on this feature: [If you don't put anything here, you are doing it wrong!]
- (replace me) I learned how to create a pull request template

### CODE REVIEW

File: app/Services/PaymentService.php [hunk 1]
```diff
@@ -38,7 +38,15 @@ public static function logPayment($data, $model)
     public static function logLateFee($data)
     {
         try {
-            $payment = LateFee::updateOrCreate(['id' => $data['id']], ['date_paid' => $data['date_paid']]);
+            if (isset($data['id'])) {
+
+                $payment = LateFee::updateOrCreate(['id' => $data['id']], ['date_paid' => $data['date_paid'], 'amount' => $data->amount]);
+            } else {
+                // dd('i got here');
```
Comment (teebarg) [inline app/Services/PaymentService.php:45]: let's remove this

File: app/Services/PaymentService.php [hunk 2]
```diff
@@ -38,7 +38,15 @@ public static function logPayment($data, $model)
...
+
+                $payment = LateFee::updateOrCreate($data);
```
Comment (teebarg) [inline app/Services/PaymentService.php:47]: why are we re-running lines 47 and 49 after 43?

File: app/Console/Commands/GenerateLateFeeCommand.php [hunk 3]
```diff
@@ -15,7 +15,7 @@ class GenerateLateFeeCommand extends BaseCommand
      *
      * @var string
      */
-    protected $signature = 'make:late-fee' . '{--day= : Generate late fee for a specific day of the month in the past e.g 18. Must be in range 1 - 31} ';
+    protected $signature = 'make:late-fee-gen' . '{--day= : Generate late fee for a specific day of the month in the past e.g 18. Must be in range 1 - 31} ';
```
Comment (teebarg) [inline app/Console/Commands/GenerateLateFeeCommand.php:18]: why are we changing this again?

File: app/Services/PaymentService.php (hunk 2 above)
Comment (darmhoo) [reply app/Services/PaymentService.php:47]: it is working for both the creating late-fee and updating late fee

Comment (darmhoo) [reply app/Services/PaymentService.php:47]: mistake

File: app/Console/Commands/GenerateLateFeeCommand.php (hunk 3 above)
Comment (darmhoo) [reply app/Console/Commands/GenerateLateFeeCommand.php:18]: it is clashing with make:late-fee-debit
one is to generate, other is to debit

**Review by teebarg — APPROVED** [review]

**Pull request merged**